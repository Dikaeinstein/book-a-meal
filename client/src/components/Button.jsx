import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ value, className }) => (
  <button className={className}>
    {value}
  </button>
);

Button.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default Button;
