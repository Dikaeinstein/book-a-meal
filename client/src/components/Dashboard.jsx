import React from 'react';
import OrderHistory from './OrderHistory';

const Dashboard = () => <h2>Dashboard</h2>;

export default Dashboard;
