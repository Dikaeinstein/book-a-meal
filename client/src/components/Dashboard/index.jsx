import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DashboardOrderHistory from './DashboardOrderHistory';
import Footer from '../util/Footer';
import FilterOrders from './FilterOrders';
import TotalOrders from './TotalOrders';
import TotalSales from './TotalSales';
import errorHandler from '../util/errorHandler';


const ConnectedDashboard = ({ fetchOrdersError }) => {
  const DashboardOrderHistoryWithErrorHandling =
    errorHandler(DashboardOrderHistory, 'Error fetching orders');

  return (
    <div>
      <main
        className="dash-main text-center bg-dark-light"
        style={{ minHeight: 'calc(100vh - 151px)' }}
      >
        <h1>Dashboard</h1>
        <section className="card dashboard">
          <TotalOrders />
          <TotalSales />
          <FilterOrders />
          <DashboardOrderHistoryWithErrorHandling
            error={fetchOrdersError}
          />
        </section>
      </main>
      <Footer />
    </div>
  );
};

ConnectedDashboard.propTypes = {
  /* eslint react/require-default-props: 0 */
  fetchOrdersError: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.objectOf(PropTypes.string),
  ]),
};

const mapStateToProps = state => ({
  fetchOrdersError: state.orders.error,
});

const Dashboard = connect(mapStateToProps)(ConnectedDashboard);

export default Dashboard;
