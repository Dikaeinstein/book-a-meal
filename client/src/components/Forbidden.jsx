import React from 'react';
import { NavLink } from 'react-router-dom';
import Footer from './util/Footer';

const Forbidden = () => (
  <main>
    <div
      className="forbidden-overlay"
      style={{
        minHeight: 'calc(100vh - 145px)',
        padding: '1rem',
      }}
    >
      <h6
        className="text-center"
        style={{ paddingTop: '60vh' }}
      >
        Sorry, access to this page is denied.
        Either check the url or <NavLink to="/">go home</NavLink>
      </h6>
    </div>
    <Footer />
  </main>
);

export default Forbidden;
