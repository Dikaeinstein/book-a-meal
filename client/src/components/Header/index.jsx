import React from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.min.css';
import Logo from './Logo';
import Navigation from './Navigation';

const Header = ({ className }) => (
  <header className={className}>
    <ToastContainer />
    <Logo
      text="Book-A-Meal"
      className="col-1-4 logo text-left text-dark"
    />
    <div className="toggle-show">&#9776;</div>
    <Navigation />
  </header>
);

Header.propTypes = {
  className: PropTypes.string.isRequired,
};

export default Header;
