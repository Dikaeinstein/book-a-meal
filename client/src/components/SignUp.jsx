import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Formik } from 'formik';
import SignUpForm from './SignUpForm';
import SideImage from './SideImage';
import userSchema from '../helpers/userSchema';
import { userSignUp } from '../actions/userActions';

const ConnectedSignUp = ({ signUp }) => {
  const handleSubmit = (values, actions) => {
    signUp(values, actions);
  };
  return (
    <main className="flex-container">
      <SideImage id="signUp" />
      <section id="signIn" className="col-1-4 aside">
        <Formik
          initialValues={{
            name: '',
            email: '',
            password: '',
            confirmPassword: '',
            role: 'customer',
          }}
          validationSchema={userSchema}
          component={SignUpForm}
          onSubmit={handleSubmit}
        />
      </section>
    </main>
  );
};

ConnectedSignUp.propTypes = {
  signUp: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch, ownProps) => ({
  signUp: (values, actions) => dispatch(userSignUp(values, actions, ownProps.history)),
});

const SignUp = connect(null, mapDispatchToProps)(ConnectedSignUp);

export default SignUp;
