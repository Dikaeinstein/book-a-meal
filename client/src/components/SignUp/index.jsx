import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Formik } from 'formik';
import SignUpForm from './SignUpForm';
import SideImage from '../util/SideImage';
import userSignupSchema from '../../validation/userSignupSchema';
import { userSignUp } from '../../actions/userActions';

const ConnectedSignUp = ({ signUp }) => {
  const handleSubmit = (values, actions) => {
    signUp(values, actions);
  };
  return (
    <main className="flex-container">
      <SideImage id="signUp" />
      <section id="signIn" className="col-1-4 aside bg-dark">
        <Formik
          initialValues={{
            name: '',
            email: '',
            password: '',
            confirmPassword: '',
            role: 'customer',
          }}
          validationSchema={userSignupSchema}
          component={SignUpForm}
          onSubmit={handleSubmit}
        />
      </section>
    </main>
  );
};

ConnectedSignUp.propTypes = {
  signUp: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch, ownProps) => ({
  signUp: (values, actions) => dispatch(userSignUp(values, actions, ownProps.history)),
});

const SignUp = connect(null, mapDispatchToProps)(ConnectedSignUp);

export default SignUp;
