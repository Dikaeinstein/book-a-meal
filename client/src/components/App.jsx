import React, { Component } from 'react';
import Header from './Header';
import IntroBanner from './IntroBanner';
import MenuCard from './MenuCard';
import Footer from './Footer';

const App = () => (
  <div>
    <Header className="navbar" />
    <IntroBanner />
		<MenuCard />
    <Footer />
  </div>
);

export default App;
