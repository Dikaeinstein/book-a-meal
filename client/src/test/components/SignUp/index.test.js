import React from 'react';
import { shallow } from 'enzyme';
import { Formik } from 'formik';
import { SignUp } from '../../../components/SignUp';


describe('<SignUp />', () => {
  const props = { signUp: jest.fn() };
  let tree;
  beforeAll(() => {
    tree = shallow(<SignUp {...props} />);
  });
  it('should call handleSubmit when form is submitted', () => {
    tree.find(Formik).simulate('submit');
    expect(props.signUp).toHaveBeenCalled();
  });
  it('should match snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
});
