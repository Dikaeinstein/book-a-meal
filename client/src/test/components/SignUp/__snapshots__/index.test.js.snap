// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> should match snapshot 1`] = `
<main
  className="flex-container"
>
  <SideImage
    id="signUp"
  />
  <section
    className="col-1-4 aside bg-dark"
    id="signIn"
  >
    <Formik
      component={[Function]}
      enableReinitialize={false}
      initialValues={
        Object {
          "confirmPassword": "",
          "email": "",
          "name": "",
          "password": "",
          "role": "customer",
        }
      }
      isInitialValid={false}
      onSubmit={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
      validationSchema={
        ObjectSchema {
          "_blacklist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "_conditions": Array [],
          "_defaultDefault": [Function],
          "_deps": Array [],
          "_excludedEdges": Array [],
          "_exclusive": Object {},
          "_mutate": false,
          "_nodes": Array [
            "role",
            "confirmPassword",
            "password",
            "email",
            "name",
          ],
          "_options": Object {
            "abortEarly": true,
            "recursive": true,
          },
          "_type": "object",
          "_typeError": [Function],
          "_whitelist": RefSet {
            "list": Set {},
            "refs": Map {},
          },
          "fields": Object {
            "confirmPassword": StringSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {},
              "_mutate": false,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "string",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {
                  null,
                },
                "refs": Map {
                  "password" => Reference {
                    "_get": [Function],
                    "isContext": false,
                    "isSelf": false,
                    "key": "password",
                    "map": [Function],
                    "path": "password",
                    "prefix": "$",
                  },
                },
              },
              "_whitelistError": [Function],
              "tests": Array [],
              "transforms": Array [
                [Function],
              ],
            },
            "email": StringSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "required": false,
                "undefined": false,
              },
              "_mutate": false,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "string",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
                [Function],
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
            },
            "name": StringSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "max": true,
                "min": true,
                "required": false,
              },
              "_mutate": false,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "string",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
            },
            "password": StringSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "max": true,
                "min": true,
                "required": false,
              },
              "_mutate": false,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "string",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
            },
            "role": StringSchema {
              "_blacklist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "_conditions": Array [],
              "_deps": Array [],
              "_exclusive": Object {
                "required": false,
                "undefined": false,
              },
              "_mutate": false,
              "_options": Object {
                "abortEarly": true,
                "recursive": true,
              },
              "_type": "string",
              "_typeError": [Function],
              "_whitelist": RefSet {
                "list": Set {},
                "refs": Map {},
              },
              "tests": Array [
                [Function],
                [Function],
                [Function],
              ],
              "transforms": Array [
                [Function],
              ],
            },
          },
          "tests": Array [],
          "transforms": Array [
            [Function],
          ],
        }
      }
    />
  </section>
</main>
`;
